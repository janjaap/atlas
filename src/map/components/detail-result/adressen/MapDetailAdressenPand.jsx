import React from 'react';
import PropTypes from 'prop-types';

import MapDetailResultItem from '../MapDetailResultItem';
import MapDetailResultStatusItem from '../MapDetailResultStatusItem';
import MapDetailResultWrapper from '../MapDetailResultWrapper';

const statusToCssModifier = {
  24: 'info',
  25: 'info',
  26: 'info',
  27: 'info',
  28: 'info',
  29: 'info',
  32: 'info'
};

const MapDetailAdressenPand = ({ panoUrl, pand, onMaximize, onPanoPreviewClick }) => (
  <MapDetailResultWrapper
    panoUrl={panoUrl}
    onMaximize={onMaximize}
    onPanoPreviewClick={onPanoPreviewClick}
    subTitle={pand.label}
    title="Pand"
  >
    <ul className="map-detail-result__list">
      <MapDetailResultItem
        label="Oorspronkelijk bouwjaar"
        value={pand.year || 'onbekend'}
      />
      <MapDetailResultStatusItem
        label="Status"
        value={pand.status.description}
        status={statusToCssModifier[pand.status.code]}
      />
    </ul>
  </MapDetailResultWrapper>
);

MapDetailAdressenPand.propTypes = {
  pand: PropTypes.shape({
    label: PropTypes.string,
    status: PropTypes.shape({
      description: PropTypes.string,
      code: PropTypes.string
    }).isRequired,
    year: PropTypes.string
  }).isRequired,
  panoUrl: PropTypes.string.isRequired,
  onMaximize: PropTypes.func.isRequired,
  onPanoPreviewClick: PropTypes.func.isRequired
};

export default MapDetailAdressenPand;
