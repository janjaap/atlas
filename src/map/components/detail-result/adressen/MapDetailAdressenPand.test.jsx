import React from 'react';
import { shallow } from 'enzyme';

import MapDetailAdressenPand from './MapDetailAdressenPand';

describe('MapDetailAdressenPand', () => {
  it('should render everything', () => {
    const pand = {
      label: 'Pand label',
      status: {
        ommschrijving: 'Pand status',
        code: '31'
      },
      year: '2020'
    };
    const clickHandler = jest.fn();
    const wrapper = shallow(
      <MapDetailAdressenPand
        panoUrl="panoUrl"
        onMaximize={clickHandler}
        onPanoPreviewClick={clickHandler}
        pand={pand}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render unknown for empty year', () => {
    const pand = {
      label: 'Pand label',
      status: {
        ommschrijving: 'Pand status',
        code: '31'
      },
      year: ''
    };
    const clickHandler = jest.fn();
    const wrapper = shallow(
      <MapDetailAdressenPand
        panoUrl="panoUrl"
        onMaximize={clickHandler}
        onPanoPreviewClick={clickHandler}
        pand={pand}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render unknown without year', () => {
    const pand = {
      label: 'Pand label',
      status: {
        ommschrijving: 'Pand status',
        code: '31'
      }
    };
    const clickHandler = jest.fn();
    const wrapper = shallow(
      <MapDetailAdressenPand
        panoUrl="panoUrl"
        onMaximize={clickHandler}
        onPanoPreviewClick={clickHandler}
        pand={pand}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
