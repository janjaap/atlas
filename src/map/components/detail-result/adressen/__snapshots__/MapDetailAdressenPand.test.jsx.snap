// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapDetailAdressenPand should render everything 1`] = `
<MapDetailResultWrapper
  onMaximize={[MockFunction]}
  onPanoPreviewClick={[MockFunction]}
  panoUrl="panoUrl"
  subTitle="Pand label"
  title="Pand"
>
  <ul
    className="map-detail-result__list"
  >
    <MapDetailResultItem
      label="Oorspronkelijk bouwjaar"
      value="2020"
    />
    <MapDetailResultStatusItem
      label="Status"
      status=""
      value=""
    />
  </ul>
</MapDetailResultWrapper>
`;

exports[`MapDetailAdressenPand should render unknown for empty year 1`] = `
<MapDetailResultWrapper
  onMaximize={[MockFunction]}
  onPanoPreviewClick={[MockFunction]}
  panoUrl="panoUrl"
  subTitle="Pand label"
  title="Pand"
>
  <ul
    className="map-detail-result__list"
  >
    <MapDetailResultItem
      label="Oorspronkelijk bouwjaar"
      value="onbekend"
    />
    <MapDetailResultStatusItem
      label="Status"
      status=""
      value=""
    />
  </ul>
</MapDetailResultWrapper>
`;

exports[`MapDetailAdressenPand should render unknown without year 1`] = `
<MapDetailResultWrapper
  onMaximize={[MockFunction]}
  onPanoPreviewClick={[MockFunction]}
  panoUrl="panoUrl"
  subTitle="Pand label"
  title="Pand"
>
  <ul
    className="map-detail-result__list"
  >
    <MapDetailResultItem
      label="Oorspronkelijk bouwjaar"
      value="onbekend"
    />
    <MapDetailResultStatusItem
      label="Status"
      status=""
      value=""
    />
  </ul>
</MapDetailResultWrapper>
`;
